### PR TITLE
[6361][4.2.x] Improve XB locking & bg modification awareness

### PR DESCRIPTION
**Pull request by jvega190**
Implement better XB awareness for background changes of the items in Preview and improve lock awareness and displaying

https://github.com/craftercms/craftercms/issues/6361

### CODE REVIEW

File: ui/guest/src/react/ExperienceBuilder.tsx [hunk 1]
```diff
@@ -163,7 +166,16 @@ function ExperienceBuilderInternal(props: InternalGuestProps) {
...
+  const {
+    editMode,
+    highlightMode,
+    editModePadding,
+    status,
+    hostCheckedIn: hasHost,
+    draggable,
+    authoringBase,
+    activeSite
+  } = state;
```
Comment (rart) [inline ui/guest/src/react/ExperienceBuilder.tsx]: I think this was a miss: `activeSite` is not in use.
```suggestion
const { editMode, highlightMode, editModePadding, status, hostCheckedIn: hasHost, draggable, authoringBase } = state;
```

File: ui/guest/src/react/ZoneMarker.tsx [hunk 2]
```diff
@@ -117,11 +133,27 @@ export function ZoneMarker(props: ZoneMarkerProps) {
...
+            {isLockedItem && (
+              <Typography noWrap variant="body2" component="div">
+                <FormattedMessage defaultMessage="Locked by {username}" values={{ username: lockInfo.username }} />
+              </Typography>
+            )}
+            {isStale && (
+              <Typography noWrap variant="body2" component="div">
+                <FormattedMessage defaultMessage="Item was modified. Refresh to enable editing." />
+              </Typography>
+            )}
```
Comment (rart) [inline ui/guest/src/react/ZoneMarker.tsx]: FormatJS is not setup on XB. Does this even work? Doesn't produce runtime exceptions?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/react/ExperienceBuilder.tsx [hunk 3]
```diff
@@ -107,6 +108,8 @@ import { setJwt } from '@craftercms/studio-ui/utils/auth';
 import { SHARED_WORKER_NAME } from '@craftercms/studio-ui/utils/constants';
 import useUnmount from '@craftercms/studio-ui/hooks/useUnmount';
 import { DeepPartial } from '@craftercms/studio-ui/models/DeepPartial';
+import { emitSystemEvent, openSiteSocket } from '@craftercms/studio-ui/state/actions/system';
```
Comment (rart) [inline ui/guest/src/react/ExperienceBuilder.tsx]: `openSiteSocket` not used
```suggestion
import { emitSystemEvent } from '@craftercms/studio-ui/state/actions/system';
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/react/ZoneMarker.tsx (hunk 2 above)
Comment (jvega190) [reply ui/guest/src/react/ZoneMarker.tsx]: I didn't see exceptions, reverted

**Review by rart — APPROVED** [review]

**Pull request merged**